uniffi_bindgen/src/bindings/swift/templates/RustBufferHelper.swift

```diff
@@ -163,24 +163,32 @@ extension ViaFfiUsingByteBuffer {
 // Implement our protocols for the built-in types that we use.
 
 extension String: ViaFfi {
-    typealias FfiType = UnsafeMutablePointer<CChar>
+    typealias FfiType = RustBuffer
 
     static func lift(_ v: FfiType) throws -> Self {
         defer {
             try! rustCall(InternalError.unknown()) { err in
-                {{ ci.ffi_string_free().name() }}(v, err)
+                {{ ci.ffi_rustbuffer_free().name() }}(v, err)
             }
         }
-        return String(cString: v)
+        if v.data == nil {
+            return String()
+        }
+        return String(bytesNoCopy: v.data!, length: Int(v.len), encoding: String.Encoding.utf8, freeWhenDone: false)!
```

Comment:
Hrm. You know, I pored over this for a good long while trying to convince myself that it would in fact copy out the bytes, but I've now forgotten why that was. And it's possible that I was delirious anyway, because it really does say "no copy" right there in the name of the thing...I'll take a fresh look at this, thanks!

Reply:
Updated to use the (IIUC) copying version, @linacambridge could you please take another look for completeness?

Reply:
Perfect!